Overwrite language switcher

katzmo opened the pull request.

katzmo committed:
Coding style fixes

katzmo committed:
Overwrite campaignion language switcher.

torotil commented on template.php:
```diff
@@ -169,6 +199,15 @@ function campaignion_foundation_block_view_alter(&$data, $block) {
     }
   }
+  // Remove disabled links from campaignion language switcher blocks.
+  if ($block->module == 'campaignion_language_switcher') {
```
I think this should be possible in `campaignion_foundation_preprocess_campaignion_language_switcher()` as well.

katzmo replied:
Unfortunately it is not, the links all still have a `href` attribute it the preprocess function, that gets only later removed by `hook_block_view()`.

torotil replied:
Are you sure?

AFAIK:  `campaignion_language_switcher_block_view()` calls `language_negotiation_get_switch_links()` which in turn invokes `hook_language_switch_links_alter()` thereby also `campaignion_language_switcher_language_switch_links_alter()` which removes the `href`. The result of `campaignion_language_switcher_block_view()` is a renderable array so the theme function is not removed until after the function has returned. So the `href` should already be removed when the theme-layer becomes active.

katzmo replied:
Yes. After `campaignion_language_switcher_block_view()` we have the final renderable array as you described, that’s why `campaignion_foundation_block_view_alter()` works. The theme function does its thing before any of that, not after as I would have expected too.

torotil replied:
That would be crazy. Are you sure? It would mean that `theme()` is run before `render()` is run on the same data structure.

katzmo replied:
It looks that way.

katzmo committed:
Render language switcher only when it has links

katzmo committed:
Fix language switcher when used with geoip

If GeoIP is enabled but there is no translation of the node for the
redirected language, some other translation will be rendered without
changing the path, so the language switcher can’t identify which node
it is displayed on (no link matches the current path).
It still needs to render an “active” link as parent item.

katzmo committed:
Add language switcher attributes

Open dropdown on click, not hover.

katzmo committed:
Adapt to new changes in campaignion_language_switcher

No more need to remove links from the list, instead `links_accessible` is
provided by the module now.

torotil reviewed (approved):
Minor thing: The comment to `campaignion_foundation_js_alter()` is not entirely accurate, but never mind.
